PracticeGit/test.c

```diff
@@ -0,0 +1,125 @@
+// -------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// -------------------------------------------------------------------------------------------------
+// test.c - Function that tests the program and main.
+// ------------------------------------------------------------------------------------------------
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <limits.h>
+#include "Program.h"
+
+#define RED     "\033[1;31m"
+#define GREEN   "\033[1;32m"
+#define CYAN    "\033[1;36m"
+#define RESET   "\033[0m"
+
+/// <summary>Execute palindrome test cases </summary>
+void RunPalindromeTest ();
+
+/// <summary>Execute Integer Reversal test cases </summary>
+void RunIntegerReversalTest ();
+
+///  <summary>Get user input for Palindrome checker </summary>
+void GetPalindromeInput ();
+
+///  <summary>Get user input for Integer reversal </summary>
+void GetIntegerInput ();
+
+void RunPalindromeTest () {
+   const char* testStrings[] = { "Eva, can I stab bats in a cave?", "malaya&lam", "Hey!! World", "J", "No lemon, no melon","$$$",
+                                 "Her favorite number is 2", "    ra   c eca  r " };
+   int expectedResults[] = { PALINDROME, PALINDROME, NOT_PALINDROME, PALINDROME, PALINDROME,NOT_PALINDROME, NOT_PALINDROME, PALINDROME };
+   int numTestCases = sizeof (testStrings) / sizeof (testStrings[0]);
+   printf ("\n                           EXECUTE PALINDROME TEST CASES \n"
+      "+------------------------------------------------------+-----------------------+----------+ \n"
+      CYAN"|                      Test String                     |    Palindrome Check   |  Result  |\n"RESET
+      "+------------------------------------------------------+-----------------------+----------+\n");
+   for (int i = 0; i < numTestCases; i++) {
+      int result = IsPalindrome_s (testStrings[i]);
+      printf ("| %-52s | %-21s | %-19s |\n", testStrings[i], result == PALINDROME ? "Palindrome" : "Not a Palindrome", result == expectedResults[i] ? GREEN "Pass" RESET : RED "Fail" RESET);
```

Comment:
Since the type of result and result == PALINDROME is the same, you can directly use result as a check.
```suggestion
      int result = IsPalindrome_s (testStrings[i]);
      printf ("| %-52s | %-21s | %-19s |\n", testStrings[i], result ? "Palindrome" : "Not a Palindrome", result == expectedResults[i] ? GREEN "Pass" RESET : RED "Fail" RESET);
```